plugins/org.jkiss.dbeaver.ui.ai/src/org/jkiss/dbeaver/ui/ai/engine/copilot/CopilotConfigurator.java

```diff
@@ -250,7 +257,14 @@ private String acquireAccessToken(@NotNull DBRProgressMonitor monitor, @NotNull
     @Override
     public Optional<AIEngineProperties> getCurrentProperties() {
         CopilotProperties copilotPropertiesCopy = new CopilotProperties();
-        saveSettings(copilotPropertiesCopy);
+        saveSettings((PROPERTIES) copilotPropertiesCopy);
         return Optional.of(copilotPropertiesCopy);
     }
+
+    protected <E extends CopilotCompletionEngine> E createEngine() {
+        CopilotProperties properties = new CopilotProperties();
+        properties.setToken(accessToken);
+
+        return (E) new CopilotCompletionEngine(properties);
```

Comment:
Redundant generic bound

```suggestion
    @NotNull
    protected CopilotCompletionEngine E createEngine() {
        CopilotProperties properties = new CopilotProperties();
        properties.setToken(accessToken);

        return new CopilotCompletionEngine(properties);
```